ibis/backends/dask/tests/test_udf.py

```diff
@@ -229,6 +230,27 @@ def test_udaf_groupby(t2, df2):
     tm.assert_frame_equal(result[columns], expected[columns])
 
 
+def test_udaf_groupby_multikey(t2, df2):
+    expr = t2.groupby([t2.key, t2.d]).aggregate(my_corr=my_corr(t2.a, t2.b))
+
+    result = expr.execute().sort_values('key').reset_index()
+
+    dfi = df2.set_index('key').compute()
+    expected = pd.DataFrame(
+        {
+            'key': list('def'),
+            'd': list('aad'),
+            'my_corr': [
+                dfi.loc[value, 'a'].corr(dfi.loc[value, 'b'])
+                for value in 'def'
+            ],
+        }
+    )
+
+    columns = ['key', 'd', 'my_corr']
+    tm.assert_frame_equal(result[columns], expected[columns])
```

Comment:
why are these not exactly equal?

Reply:
Great point - partially something I inherited from the pandas tests https://github.com/gerrymanoim/ibis/blob/ccdc6ddeb388a3c49417c7adccecd45b3c923e87/ibis/backends/pandas/tests/test_udf.py#L216-L216 (I don't think the pandas tests need to do this either), partially resetting the index without dropping it. 

I've fixed this and updated the tests.